### ISSUE TITLE
Clean code in loss calculation.

### ISSUE THREAD

**Issue by tyui592** [fundamentalvision/Deformable-DETR#243]
https://github.com/fundamentalvision/Deformable-DETR/blob/11169a60c33333af00a4849f1808023eba96a931/models/deformable_detr.py#L233-L237

In 233 line, if we don't add 1 to src_logits.shape[2], we don't need to reduce it to [:,:,:-1] in line 237.

**Comment by tyui592** [comment]
It was my misunderstanding about num_class. Sorry.
